ext/standard/php_image.h

```diff
@@ -54,4 +55,13 @@ PHPAPI int php_getimagetype(php_stream *stream, const char *input, char *filetyp
 
 PHPAPI char * php_image_type_to_mime_type(int image_type);
 
+/* Struct and function to be used for testing whether an image is AVIF */
+typedef struct {
+  php_stream * stream;
+  zend_string * data;
+  size_t data_pos;
+} php_gd_image_reader;
```

Comment:
I'd call that just `php_image_reader` since it's not about GD (GD is only a client). Also, it seems to me that either `stream` or `data` is used, so this could be a union; we'd need a tag in this case.